macros/edr/data_monitoring/anomaly_detection/construct_metric_properties_dict.sql

```diff
@@ -0,0 +1,7 @@
+{% macro construct_metric_properties_dict(timestamp_column, where_expression, time_bucket) %}
```

Comment:
Why not?
```jinja
{% set d = {"timestamp_column": timestamp_column, "where_expression": where_expression, "time_bucket": time_bucket} %}
```
I think that changing it to a `Null` string is dangerous and could lead to bugs.

Reply:
moving to a function instead of duplicating seemed cleaner.

The null string (`"Null"`) is to avoid jsonSerializationError when calling json_serialize on a dict with a `None` value.

Reply:
But it's not an error.
`json.dumps()` (`tojson()`) converts `None` to `null`.

```py
ipdb> ctx = locals()['context']
ipdb> ctx["tojson"]({"hello": None})
'{"hello": null}'
```

Reply:
Figured out that the issue was not with json_serializing None, but `Undefined` . Fixed the code to convert Undefined to None, now it works and the string "Null" isn't participating in the code any more.